addons/hr_holidays/models/hr_leave_type.py

```diff
@@ -479,6 +480,13 @@ def get_allocation_data(self, employees, target_date=None):
                             continue
                         if allocation.date_to and allocation.date_to < target_date:
                             continue
+                    # If the unit for allocation is not the same as the request, convert data to the appropriate unit
+                    display_unit = 'hour' if leave_type.request_unit == 'hour' else 'day'
+                    if data['allocation_unit'] != display_unit:
+                        for key in ['virtual_leaves_taken', 'leaves_taken', 'virtual_remaining_leaves',
+                                    'remaining_leaves', 'max_leaves', 'accrual_bonus']:
+                            amount = data[key]
+                            data[key] = employee._convert_time_off_amount(amount, data['allocation_unit'], display_unit)
```

Comment:
key is not that readable IMO given the size of the method and the global complexity of it

Reply:
> key is not that readable IMO given the size of the method and the global complexity of it

What do you propose to use here? As we are iterating over a list of keys we need something generic here, and I can't think of anything better 😅